app/src/main/java/com/willowtree/vocable/presets/adapter/AddToCategoryPickerViewModel.kt

```diff
@@ -0,0 +1,158 @@
+package com.willowtree.vocable.presets.adapter
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.willowtree.vocable.BaseViewModel
+import com.willowtree.vocable.keyboard.KeyboardViewModel
+import com.willowtree.vocable.presets.PresetsRepository
+import com.willowtree.vocable.room.Category
+import com.willowtree.vocable.room.CategoryPhraseCrossRef
+import com.willowtree.vocable.room.Phrase
+import com.willowtree.vocable.utils.LocalizedResourceUtility
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.koin.core.inject
+import java.util.*
+
+class AddToCategoryPickerViewModel : BaseViewModel() {
+
+    private val presetsRepository: PresetsRepository by inject()
+    private val localizedResourceUtility: LocalizedResourceUtility by inject()
+
+    private val liveCategoryMap = MutableLiveData<Map<Category, Boolean>>()
+    val categoryMap: LiveData<Map<Category, Boolean>> = liveCategoryMap
+
+    private val liveShowPhraseAdded = MutableLiveData<Boolean>()
+    val showPhraseAdded: LiveData<Boolean> = liveShowPhraseAdded
+
+    private val liveShowPhraseDeleted = MutableLiveData<Boolean>()
+    val showPhraseDeleted: LiveData<Boolean> = liveShowPhraseDeleted
+
+    private var userCategories = listOf<Category>()
+
+    // a list that maps categories to the phrase.  The phrase will not change, so we just need
+    // to map the category id to the phrase object that was added to it, so we can delete it later
+    private var savedPhrases = mutableMapOf<String, Phrase>()
+
+    fun handleCategoryToggled(phraseString: String, category: Category, isChecked: Boolean) {
+        backgroundScope.launch {
+            // if the user checked the switch
+            if (isChecked) {
+                // add it to the category
+                addNewPhrase(phraseString, category)
+            } else { // if they un-checked the switch
+                // delete it from the category
+                deletePhraseFromCategory(phraseString, category)
+            }
+        }
+    }
+
+    fun buildCategoryList(phraseString: String) {
+        // map category to whether or not the phrase is in that category
+        val phraseMap = mutableMapOf<Category, Boolean>()
+
+        backgroundScope.launch {
+            userCategories = presetsRepository.getUserGeneratedCategories()
+
+            // for each category in the list
+            for (category in userCategories) {
+                // get all the phrases associated with it
+                val crossRefs = presetsRepository.getCrossRefsForCategoryId(category.categoryId)
+
+                // for each phrase in the category
+                for (ref in crossRefs) {
+                    // get the phrase from the db
+                    val phrase = presetsRepository.getPhraseById(ref.phraseId)
```

Comment:
There's already a call to get all the phrases for a given category - `presetsRepository.getPhrasesForCategory(categoryId: String)`